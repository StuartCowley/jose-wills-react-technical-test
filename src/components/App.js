import React from "react";
import "../styles/app.css";
import Search from "./Search";

const App = () => {
  return (
    <div className="app">
      <img className="app__nasa-logo" src={require("../images/nasa-worm-logo.jpg")} alt="nasa-logo" />
      <Search />
    </div>
  );
};

export default App;
