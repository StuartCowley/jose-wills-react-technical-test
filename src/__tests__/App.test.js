import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
    it("renders correctly", () => {
        const { asFragment } = render(<App />);

        expect(asFragment()).toMatchSnapshot();
    });

    it("uses the right source for image", () => {
        render(<App />);
        const image = screen.getByAltText("nasa-logo");

        expect(image.src).toContain(require("../images/nasa-worm-logo.jpg"));
    });

    it("renders the image", () => {
        render(<App />);
        const image = screen.getByAltText("nasa-logo");

        expect(image).toBeInTheDocument();
    });
});