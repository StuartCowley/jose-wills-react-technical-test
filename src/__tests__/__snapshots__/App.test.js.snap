// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <img
      alt="nasa-logo"
      class="app__nasa-logo"
      src="nasa-worm-logo.jpg"
    />
    <input
      class="search__input"
      placeholder="Enter image keywords here"
      type="search"
    />
  </div>
</DocumentFragment>
`;
